operator/internal/manifests/ingester.go

```diff
@@ -274,3 +277,30 @@ func configureIngesterGRPCServicePKI(sts *appsv1.StatefulSet, opts Options) erro
 	serviceName := serviceNameIngesterGRPC(opts.Name)
 	return configureGRPCServicePKI(&sts.Spec.Template.Spec, serviceName)
 }
+
+// newIngesterPodDisruptionBudget returns a PodDisruptionBudget for the LokiStack
+// Ingester pods.
+func newIngesterPodDisruptionBudget(opts Options) *policyv1.PodDisruptionBudget {
+	l := ComponentLabels(LabelIngesterComponent, opts.Name)
+	mu := intstr.FromInt(1)
+	if opts.Stack.Size == v1.SizeOneXMedium {
+		mu = intstr.FromInt(2)
+	}
```

Comment:
To avoid having endless switch cases inline here when growing in supported deployment sizes, we have a central registry for such constants: https://github.com/grafana/loki/blob/main/operator/internal/manifests/internal/sizes.go

Reply:
So IIUC I should add a field to `ResourceRequirements` like `pdbMinAvailable` that will be set for each of the different sizes, and then in this part of the code we will simply get that value.

Reply:
Yes, spot on!